tests/system/src/test/java/test/robot/javafx/scene/TabPanePermuteGetTabsTest.java

```diff
@@ -181,6 +189,59 @@ public void testPermuteGetTabsWithMoreTabs() {
         }
     }
 
+    // Test for JDK-8237602
+    @Test
+    public void testPermutGetTabsWithMoreTabs1() {
+        // Step #1
+        Util.runAndWait(() -> {
+            tabPane.getTabs().setAll(tab[0], tab[1]);
+            tabPane.getSelectionModel().select(tab[1]);
+        });
+        delay();
+
+        Assert.assertSame("Sanity: tab[1] should be the selected tab.",
+            tab[1], tabPane.getSelectionModel().getSelectedItem());
+
+        // Step #2
+        Util.runAndWait(() -> {
+            tabPane.getTabs().setAll(tab[0], tab[1], tab[2], tab[3]);
+        });
+        delay();
+
+        // Step #3.1
+        Assert.assertSame("Sanity: tab[1] should remain selected tab after permuting.",
+            tab[1], tabPane.getSelectionModel().getSelectedItem());
+
+        // Step #3.2
+        // Click on first tab header
+        selectionLatch[0] = new CountDownLatch(1);
+        Util.runAndWait(() -> {
+            robot.mouseMove((int) (scene.getWindow().getX() + scene.getX() + firstTabdXY),
+                    (int) (scene.getWindow().getY() + scene.getY() + firstTabdXY));
+            robot.mousePress(MouseButton.PRIMARY);
+            robot.mouseRelease(MouseButton.PRIMARY);
+        });
+        delay();
```

Comment:
Does using Robot to select the Tab test add value? Probably not worth worrying about given that this will all move to the controls unit tests with [JDK-8244195](https://bugs.openjdk.java.net/browse/JDK-8244195).

Reply:
In these scenarios, the `TabPane.getTabs()` has correct sequence. So selecting a tab using `TabPaneSelectionModel` does not test the issue. Using Mouse guarantees that `TabHeaderArea` has correct order of tabs. But anyway once we convert these tests to unit tests, mouse events will be removed.